worker/src/core/services/phonebook.service.ts

```diff
@@ -0,0 +1,128 @@
+import config from '@core/config'
+import PhonebookClient from '@shared/clients/phonebook-client.class'
+import {
+  PhonebookChannelDto,
+  UserChannel,
+} from '@shared/clients/phonebook-client.class/interfaces'
+import { map } from 'lodash'
+import { EmailResultRow, Message } from '@core/loaders/message-worker/interface'
+import CircuitBreaker from 'opossum'
+
+const phonebookClient: PhonebookClient = new PhonebookClient(
+  config.get('phonebook.endpointUrl'),
+  config.get('phonebook.apiKey')
+)
+
+const options = {
+  timeout: 5000, // Trigger failure if phonebook takes longer than 5sec to respond
+  errorThresholdPercentage: 20, // When 20% of requests fail, open the circuit
+  resetTimeout: 30000, // After 30 seconds, half open the circuit and try again.
+  enabled: true,
+}
+
+const getUniqueLinksForUsers = async (
+  body: PhonebookChannelDto
+): Promise<Map<string, UserChannel>> => {
+  const userChannelsRes = await phonebookClient.getUniqueLinksForUsers(body)
+  return userChannelsRes.reduce(
+    (map: Map<string, UserChannel>, userChannel: UserChannel) => {
+      map.set(userChannel.channelId, userChannel)
+      return map
+    },
+    new Map<string, UserChannel>()
+  )
+}
+
+const breaker = new CircuitBreaker(getUniqueLinksForUsers, options)
+
+const appendLinkForEmail = async (
+  result: EmailResultRow[],
+  channel = 'Email'
+): Promise<Message[]> => {
+  const showMastheadDomain = config.get('showMastheadDomain')
+
+  const channels: UserChannel[] = result.map((row) => {
+    return {
+      channel,
+      channelId: row.message.recipient,
+    }
+  })
+  const payload: PhonebookChannelDto = {
+    userChannels: channels,
+  }
+
+  return breaker
+    .fire(payload)
+    .then((res) => {
```

Comment:
Can we write this in a normal async/await flow instead of `.then.catch` as this function has already been declared `async`

Reply:
async await doesnt work. The breaker code works only with then

Reply:
@richard-yak @hanu-ogp have we tried that out and confirmed that it doesnt work? Cos for this specific piece of code I don't see how it wouldn't. 
Can we tried it out and compare the behaviours? I'm really curious to why it would not work

Reply:
yes i tried last time when i implemented circuit breaker and async await didnt work. We should merge this PR as this is a non blocking and cosmetic issue and @richard-yak Can you follow up post launch to see if code can be converted to async await